qsirecon/interfaces/gradients.py

```diff
@@ -403,3 +411,28 @@ def _select_lines(in_file, out_file, indices):
 
     with open(out_file, "w") as out_f:
         out_f.writelines(new_lines)
+
+
+def _classify_shell_scheme(bval_file, max_distance=5):
+    """Classify a shell scheme into a known type."""
+    from sklearn.metrics import silhouette_score
+
+    bvals = np.loadtxt(bval_file)
+    X = bvals.reshape(-1, 1)
+    shells = _cluster_bvals(bvals, max_distance)
+    n_clusters = len(np.unique(shells))
+
+    # Check if it's a DSI protocol
+    score = silhouette_score(X, shells)
+    if score < 0.8:
+        return "spectrum"
```

Comment:
```suggestion
        return "non-shelled"
```

This can be cartesian or random (like cs-dsi)

Reply:
the user would need to provide some sort of parameter about how non-gridlike the acquisition can be before it's considered a cs-dsi. So I think non-shelled is good enough for out purposes